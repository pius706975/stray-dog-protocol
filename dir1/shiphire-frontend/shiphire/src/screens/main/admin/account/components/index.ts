export { default as AccountMenuItem } from './AccountMenuItem';
