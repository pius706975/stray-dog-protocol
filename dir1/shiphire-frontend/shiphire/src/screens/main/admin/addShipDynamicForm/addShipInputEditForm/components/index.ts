export { default as AdditionalInput } from './AdditionalInput';
export { default as SelectInputProperties } from './SelectInputProperties';
