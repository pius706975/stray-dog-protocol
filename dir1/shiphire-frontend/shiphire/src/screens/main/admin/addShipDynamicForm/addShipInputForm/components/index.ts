export { default as InputPreview } from './InputPreview';
export { default as SelectInputProperties } from './SelectInputProperties';
export { default as AdditionalInput } from './AdditionalInput';
