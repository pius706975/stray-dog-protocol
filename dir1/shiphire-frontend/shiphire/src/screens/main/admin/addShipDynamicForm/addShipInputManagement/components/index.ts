export { default as DetailInputSheet } from './DetailInputSheet';
export { default as ModalConfirmation } from './ModalConfirmation';
