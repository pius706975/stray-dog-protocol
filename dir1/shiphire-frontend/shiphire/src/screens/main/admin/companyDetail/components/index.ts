export { default as CompanyDocumentItem } from './CompanyDocumentItem';
