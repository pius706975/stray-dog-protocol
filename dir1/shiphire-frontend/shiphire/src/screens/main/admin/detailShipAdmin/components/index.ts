export { default as DetailShipSheet } from './DetailShipSheet';
export { default as DetailShipProfile } from './DetailShipProfile';
export { default as ModalPdf } from './ModalPdf';
