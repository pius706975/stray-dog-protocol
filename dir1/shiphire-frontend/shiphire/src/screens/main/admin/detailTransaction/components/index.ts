export {default as DetailTransactionField} from './DetailTransactionField';
export {default as DocumentField} from './DocumentField';