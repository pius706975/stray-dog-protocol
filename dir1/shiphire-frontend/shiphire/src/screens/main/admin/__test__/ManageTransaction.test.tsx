import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainAdminStackParamList } from '../../../../types';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';

import { ManageTransaction } from '../manageTransaction';
import { useGetAllUserTransactions } from '../../../../hooks';
import {
    fireEvent,
    render,
    userEvent,
    waitFor,
} from '@testing-library/react-native';
import renderer, { act } from 'react-test-renderer';
import { TransactionCard } from '../manageTransaction/components';

jest.mock('../../../../hooks');
const queryClient = new QueryClient();
const MainAdminStack = createNativeStackNavigator<MainAdminStackParamList>();

const mock = new MockAdapter(httpRequest);
const mockTransaction = [
    {
        _id: '1234',
        rentalId: 'SH-1234',
        renterId: '456',
        rentalDuration: 2,
        ship: {
            name: 'Ship 1',
        },
        status: [
            {
                name: 'rfq 1',
                desc: 'RFQ Sent',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
            {
                name: 'rfq 2',
                desc: 'RFQ Accepted',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
        ],
        ownerDetails: {
            company: {
                name: 'test company',
            },
        },
        payment: [
            {
                _id: '659634e1d7911740a0d61b41',
                paymentReminded: [],
                createdAt: new Date('2024-01-04T04:33:12.268Z'),
                paymentApproved: true,
                paymentId: 'PY-01042024-a3029559',
                receiptUrl: 'https://storage.googleapis.com',
            },
            {
                createdAt: new Date('2024-01-04T04:42:33.655Z'),
                _id: '659635bad7911740a0d61caa',
                paymentReminded: [],
                paymentApproved: false,
                paymentId: 'PY-01042024-bf706fd0',
                receiptUrl: 'https://storage.googleapis.com',
            },
        ],
    },
    {
        _id: '4567',
        rentalId: 'SH-4567',
        renterId: '456',
        rentalDuration: 2,
        ship: {
            name: 'Barge Hauler',
        },
        status: [
            {
                name: 'rfq 1',
                desc: 'RFQ Sent',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
            {
                name: 'rfq 2',
                desc: 'RFQ Accepted',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
        ],
        ownerDetails: {
            company: {
                name: 'test company',
            },
        },
        payment: [
            {
                _id: '659634e1d7911740a0d61b41',
                paymentReminded: [],
                createdAt: new Date('2024-01-04T04:33:12.268Z'),
                paymentApproved: true,
                paymentId: 'PY-01042024-a3029559',
                receiptUrl: 'https://storage.googleapis.com',
            },
            {
                createdAt: new Date('2024-01-04T04:42:33.655Z'),
                _id: '659635bad7911740a0d61caa',
                paymentReminded: [],
                paymentApproved: false,
                paymentId: 'PY-01042024-bf706fd0',
                receiptUrl: 'https://storage.googleapis.com',
            },
        ],
    },
    {
        _id: '786',
        rentalId: 'SH-4567',
        renterId: '456',
        rentalDuration: 2,
        ship: {
            name: 'MV Cruiser',
        },
        status: [
            {
                name: 'rfq 1',
                desc: 'RFQ Sent',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
            {
                name: 'rfq 2',
                desc: 'RFQ Accepted',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
        ],
        ownerDetails: {
            company: {
                name: 'test company',
            },
        },
        payment: [
            {
                _id: '659634e1d7911740a0d61b41',
                paymentReminded: [],
                createdAt: new Date('2024-01-04T04:33:12.268Z'),
                paymentApproved: true,
                paymentId: 'PY-01042024-a3029559',
                receiptUrl: 'https://storage.googleapis.com',
            },
            {
                createdAt: new Date('2024-01-04T04:42:33.655Z'),
                _id: '659635bad7911740a0d61caa',
                paymentReminded: [],
                paymentApproved: false,
                paymentId: 'PY-01042024-bf706fd0',
                receiptUrl: 'https://storage.googleapis.com',
            },
        ],
    },
    {
        _id: '3435',
        rentalId: 'SH-4567',
        renterId: '456',
        rentalDuration: 2,
        ship: {
            name: 'Tugboat',
        },
        status: [
            {
                name: 'rfq 1',
                desc: 'RFQ Sent',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
            {
                name: 'rfq 2',
                desc: 'RFQ Accepted',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
        ],
        ownerDetails: {
            company: {
                name: 'test company',
            },
        },
        payment: [
            {
                _id: '659634e1d7911740a0d61b41',
                paymentReminded: [],
                createdAt: new Date('2024-01-04T04:33:12.268Z'),
                paymentApproved: true,
                paymentId: 'PY-01042024-a3029559',
                receiptUrl: 'https://storage.googleapis.com',
            },
            {
                createdAt: new Date('2024-01-04T04:42:33.655Z'),
                _id: '659635bad7911740a0d61caa',
                paymentReminded: [],
                paymentApproved: false,
                paymentId: 'PY-01042024-bf706fd0',
                receiptUrl: 'https://storage.googleapis.com',
            },
        ],
    },
    {
        _id: '9476',
        rentalId: 'SH-4567',
        renterId: '456',
        rentalDuration: 2,
        ship: {
            name: 'Seven',
        },
        status: [
            {
                name: 'rfq 1',
                desc: 'RFQ Sent',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
            {
                name: 'rfq 2',
                desc: 'RFQ Accepted',
                isOpened: true,
                date: new Date(),
                _id: '1234',
            },
        ],
        ownerDetails: {
            company: {
                name: 'test company',
            },
        },
        payment: [
            {
                _id: '659634e1d7911740a0d61b41',
                paymentReminded: [],
                createdAt: new Date('2024-01-04T04:33:12.268Z'),
                paymentApproved: true,
                paymentId: 'PY-01042024-a3029559',
                receiptUrl: 'https://storage.googleapis.com',
            },
            {
                createdAt: new Date('2024-01-04T04:42:33.655Z'),
                _id: '659635bad7911740a0d61caa',
                paymentReminded: [],
                paymentApproved: false,
                paymentId: 'PY-01042024-bf706fd0',
                receiptUrl: 'https://storage.googleapis.com',
            },
        ],
    },
];
const adminStackMockComponent = () => (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainAdminStack.Navigator>
                    <MainAdminStack.Screen
                        name="ManageTransaction"
                        component={ManageTransaction}
                    />
                </MainAdminStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);
beforeEach(() => {
    mock.onGet('/admin/get-transaction').reply(200, {
        message: 'success',
        data: mockTransaction,
    });
    (useGetAllUserTransactions as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                data: {
                    data: mockTransaction,
                },
            };
            onSuccess(mockResponseData);
        }),
    }));
});
describe('Testing transaction management screen', () => {
    describe('Snapshot testing', () => {
        it('should render transaction management screen corectly', async () => {
            const transactionManagementScreen = renderer.create(
                adminStackMockComponent(),
            );
            expect(transactionManagementScreen).toMatchSnapshot();
        });
    });
    describe('Unit Test', () => {
        it('should render all transaction list', () => {
            const { getByTestId } = render(adminStackMockComponent());
            // rerender(adminStackMockComponent());
            const flatList = getByTestId('transaction-flatlist');

            expect(flatList.props.data.length).toBe(5);
        });
        it('should paginate on scroll', async () => {
            const { getByTestId, rerender } = render(adminStackMockComponent());
            const flatList = getByTestId('transaction-flatlist');
            const user = userEvent.setup();
            act(() => {
                fireEvent(flatList, 'onEndReached');
            });
            // await new Promise(resolve => setTimeout(resolve, 500));

            expect(flatList.props.data.length).toBe(10);
        });
        it('should render transaction list when search ship name', async () => {
            mock.onGet('/admin/get-transaction').reply(200, {
                message: 'success',
                data: [mockTransaction[0]],
            });
            (useGetAllUserTransactions as jest.Mock).mockImplementation(() => ({
                mutate: jest.fn().mockImplementation((data, options) => {
                    const { onSuccess } = options;
                    const mockResponseData = {
                        data: {
                            data: [mockTransaction[0]],
                        },
                    };
                    onSuccess(mockResponseData);
                }),
            }));

            const { getByTestId, rerender } = render(adminStackMockComponent());
            const searchBar = getByTestId('search-bar');

            act(() => {
                fireEvent.changeText(searchBar, 'Shi');
                jest.advanceTimersByTime(499);
            });

            rerender(adminStackMockComponent());

            await waitFor(() => {
                const flatList = getByTestId('transaction-flatlist');

                expect(flatList.props.data.length).toBe(1);
            });
        });
        it('should render transaction card component', () => {
            const { getByTestId } = render(
                <TransactionCard
                    testID="transaction-0"
                    onPress={jest.fn()}
                    status={mockTransaction[0].status[1]}
                    shipImage=""
                    shipName={mockTransaction[0].ship.name}
                    rentalId={mockTransaction[0].rentalId}
                    totalRent={23000}
                    shipCompany={''}
                    timeStamp={new Date()}
                />,
            );
            const pressable = getByTestId('transaction-0');
            expect('Ship 1').toBeDefined();
            expect('SH-1234').toBeDefined();
            expect('23000').toBeDefined();
            expect(pressable).toBeDefined();
        });
    });
});
