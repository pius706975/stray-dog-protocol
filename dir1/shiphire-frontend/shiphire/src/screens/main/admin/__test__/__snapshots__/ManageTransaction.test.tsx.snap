// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing transaction management screen Snapshoot testing should render transaction management screen corectly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            collapsable={false}
            data={
              [
                "",
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            handlerTag={1}
            handlerType="NativeViewGestureHandler"
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollEventThrottle={50}
            showsVerticalScrollIndicator={false}
            stickyHeaderIndices={[]}
            style={
              {
                "marginHorizontal": 20,
              }
            }
            viewabilityConfigCallbackPairs={[]}
            waitFor={
              [
                {
                  "current": null,
                },
                {
                  "current": null,
                },
              ]
            }
          >
            <View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      {
                        "flex": 1,
                      },
                      undefined,
                      {},
                      {},
                      {},
                      {
                        "backgroundColor": "#f0f6fb",
                        "justifyContent": "space-between",
                        "position": "relative",
                      },
                    ]
                  }
                  testID="adminHomeScreen"
                >
                  <RCTScrollView
                    contentContainerStyle={
                      {
                        "alignItems": undefined,
                        "flex": undefined,
                        "flexGrow": 1,
                        "gap": undefined,
                        "justifyContent": undefined,
                        "marginBottom": 40,
                        "padding": undefined,
                        "paddingVertical": undefined,
                      }
                    }
                  >
                    <View>
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#4d5963",
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 22,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        Transaction List
                      </Text>
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "backgroundColor": "#cbcdd1",
                            "borderRadius": 5,
                            "flexDirection": "row",
                            "height": 50,
                            "justifyContent": "center",
                            "marginVertical": 12,
                            "overflow": "hidden",
                            "paddingBottom": 13,
                            "paddingTop": 13,
                          }
                        }
                        testID="RNE__SearchBar-wrapper"
                      >
                        <View
                          style={
                            {
                              "paddingHorizontal": 0,
                              "width": "100%",
                            }
                          }
                          testID="RNE__Input__view-wrapper"
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "alignItems": "center",
                                "backgroundColor": "#cbcdd1",
                                "borderBottomWidth": 0,
                                "borderColor": "#86939e",
                                "borderRadius": 5,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "marginLeft": 8,
                                "marginRight": 8,
                                "minHeight": 36,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                ],
                              }
                            }
                          >
                            <View
                              style={
                                {
                                  "alignItems": "center",
                                  "height": 40,
                                  "justifyContent": "center",
                                  "marginLeft": 8,
                                  "marginVertical": 4,
                                  "paddingRight": 4,
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "overflow": "hidden",
                                  }
                                }
                                testID="RNE__ICON__CONTAINER"
                              >
                                <View
                                  accessibilityRole="button"
                                  android_ripple={
                                    {
                                      "borderless": false,
                                      "color": "rgba(0, 0, 0, 0.3)",
                                      "radius": -5,
                                    }
                                  }
                                  disabled={false}
                                  testID="RNE__ICON__CONTAINER_ACTION"
                                >
                                  <View
                                    style={
                                      {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                    testID="RNE__ICON"
                                  >
                                    <Text
                                      allowFontScaling={false}
                                      selectable={false}
                                      style={
                                        [
                                          {
                                            "color": "#7d7d7d",
                                            "fontSize": 20,
                                          },
                                          {
                                            "backgroundColor": "transparent",
                                          },
                                          {
                                            "fontFamily": "Ionicons",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          {},
                                        ]
                                      }
                                      testID="RNE__ICON__Component"
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                            <TextInput
                              editable={true}
                              onBlur={[Function]}
                              onCancel={[Function]}
                              onChangeText={[Function]}
                              onFocus={[Function]}
                              placeholder="Search Transaction"
                              placeholderTextColor="#7d7d7d"
                              platform="ios"
                              style={
                                {
                                  "color": "#4d5963",
                                  "flex": 1,
                                  "fontFamily": "Kanit-Regular",
                                  "fontSize": 20,
                                  "marginLeft": 6,
                                  "minHeight": 40,
                                  "overflow": "hidden",
                                }
                              }
                              testID="search-bar"
                              underlineColorAndroid="transparent"
                              value=""
                            />
                            <View
                              style={
                                {
                                  "alignItems": "center",
                                  "height": 40,
                                  "justifyContent": "center",
                                  "marginRight": 8,
                                  "marginVertical": 4,
                                  "paddingRight": 4,
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "flexDirection": "row",
                                  }
                                }
                              />
                            </View>
                          </View>
                          <Text
                            style={
                              {
                                "color": "#ff190c",
                                "fontSize": 12,
                                "height": 0,
                                "margin": 0,
                                "padding": 0,
                              }
                            }
                          />
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            {
                              "opacity": 0,
                              "position": "absolute",
                              "right": null,
                            }
                          }
                          testID="RNE__SearchBar-cancelButtonContainer"
                        >
                          <View
                            accessibilityRole="button"
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                          >
                            <View
                              style={{}}
                              testID="RNE__SearchBar-cancelButton"
                            >
                              <Text
                                style={
                                  {
                                    "color": "#007aff",
                                    "fontSize": 18,
                                    "padding": 8,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Cancel
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                      <RCTScrollView
                        ItemSeparatorComponent={[Function]}
                        collapsable={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        handlerTag={2}
                        handlerType="NativeViewGestureHandler"
                        keyExtractor={[Function]}
                        onContentSizeChange={[Function]}
                        onEndReached={[Function]}
                        onEndReachedThreshold={0.1}
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        onLayout={[Function]}
                        onMomentumScrollBegin={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        renderScrollComponent={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={[]}
                        testID="transaction-flatlist"
                        viewabilityConfigCallbackPairs={[]}
                        waitFor={
                          [
                            {
                              "current": null,
                            },
                            {
                              "current": null,
                            },
                          ]
                        }
                      >
                        <View />
                      </RCTScrollView>
                    </View>
                  </RCTScrollView>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="ManageTransaction"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`Testing transaction management screen Snapshot testing should render transaction management screen corectly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            collapsable={false}
            data={
              [
                "",
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            handlerTag={1}
            handlerType="NativeViewGestureHandler"
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollEventThrottle={50}
            showsVerticalScrollIndicator={false}
            stickyHeaderIndices={[]}
            style={
              {
                "marginHorizontal": 20,
              }
            }
            viewabilityConfigCallbackPairs={[]}
            waitFor={
              [
                {
                  "current": null,
                },
                {
                  "current": null,
                },
              ]
            }
          >
            <View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      {
                        "flex": 1,
                      },
                      undefined,
                      {},
                      {},
                      {},
                      {
                        "backgroundColor": "#f0f6fb",
                        "justifyContent": "space-between",
                        "position": "relative",
                      },
                    ]
                  }
                  testID="adminHomeScreen"
                >
                  <RCTScrollView
                    contentContainerStyle={
                      {
                        "alignItems": undefined,
                        "flex": undefined,
                        "flexGrow": 1,
                        "gap": undefined,
                        "justifyContent": undefined,
                        "marginBottom": 40,
                        "padding": undefined,
                        "paddingVertical": undefined,
                      }
                    }
                  >
                    <View>
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#4d5963",
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 22,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        Transaction List
                      </Text>
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "backgroundColor": "#cbcdd1",
                            "borderRadius": 5,
                            "flexDirection": "row",
                            "height": 50,
                            "justifyContent": "center",
                            "marginVertical": 12,
                            "overflow": "hidden",
                            "paddingBottom": 13,
                            "paddingTop": 13,
                          }
                        }
                        testID="RNE__SearchBar-wrapper"
                      >
                        <View
                          style={
                            {
                              "paddingHorizontal": 0,
                              "width": "100%",
                            }
                          }
                          testID="RNE__Input__view-wrapper"
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "alignItems": "center",
                                "backgroundColor": "#cbcdd1",
                                "borderBottomWidth": 0,
                                "borderColor": "#86939e",
                                "borderRadius": 5,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "marginLeft": 8,
                                "marginRight": 8,
                                "minHeight": 36,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                ],
                              }
                            }
                          >
                            <View
                              style={
                                {
                                  "alignItems": "center",
                                  "height": 40,
                                  "justifyContent": "center",
                                  "marginLeft": 8,
                                  "marginVertical": 4,
                                  "paddingRight": 4,
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "overflow": "hidden",
                                  }
                                }
                                testID="RNE__ICON__CONTAINER"
                              >
                                <View
                                  accessibilityRole="button"
                                  android_ripple={
                                    {
                                      "borderless": false,
                                      "color": "rgba(0, 0, 0, 0.3)",
                                      "radius": -5,
                                    }
                                  }
                                  disabled={false}
                                  testID="RNE__ICON__CONTAINER_ACTION"
                                >
                                  <View
                                    style={
                                      {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                    testID="RNE__ICON"
                                  >
                                    <Text
                                      allowFontScaling={false}
                                      selectable={false}
                                      style={
                                        [
                                          {
                                            "color": "#7d7d7d",
                                            "fontSize": 20,
                                          },
                                          {
                                            "backgroundColor": "transparent",
                                          },
                                          {
                                            "fontFamily": "Ionicons",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          {},
                                        ]
                                      }
                                      testID="RNE__ICON__Component"
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                            <TextInput
                              editable={true}
                              onBlur={[Function]}
                              onCancel={[Function]}
                              onChangeText={[Function]}
                              onFocus={[Function]}
                              placeholder="Search Transaction"
                              placeholderTextColor="#7d7d7d"
                              platform="ios"
                              style={
                                {
                                  "color": "#4d5963",
                                  "flex": 1,
                                  "fontFamily": "Kanit-Regular",
                                  "fontSize": 20,
                                  "marginLeft": 6,
                                  "minHeight": 40,
                                  "overflow": "hidden",
                                }
                              }
                              testID="search-bar"
                              underlineColorAndroid="transparent"
                              value=""
                            />
                            <View
                              style={
                                {
                                  "alignItems": "center",
                                  "height": 40,
                                  "justifyContent": "center",
                                  "marginRight": 8,
                                  "marginVertical": 4,
                                  "paddingRight": 4,
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "flexDirection": "row",
                                  }
                                }
                              />
                            </View>
                          </View>
                          <Text
                            style={
                              {
                                "color": "#ff190c",
                                "fontSize": 12,
                                "height": 0,
                                "margin": 0,
                                "padding": 0,
                              }
                            }
                          />
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            {
                              "opacity": 0,
                              "position": "absolute",
                              "right": null,
                            }
                          }
                          testID="RNE__SearchBar-cancelButtonContainer"
                        >
                          <View
                            accessibilityRole="button"
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                          >
                            <View
                              style={{}}
                              testID="RNE__SearchBar-cancelButton"
                            >
                              <Text
                                style={
                                  {
                                    "color": "#007aff",
                                    "fontSize": 18,
                                    "padding": 8,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Cancel
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                      <RCTScrollView
                        ItemSeparatorComponent={[Function]}
                        collapsable={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        handlerTag={2}
                        handlerType="NativeViewGestureHandler"
                        keyExtractor={[Function]}
                        onContentSizeChange={[Function]}
                        onEndReached={[Function]}
                        onEndReachedThreshold={0.1}
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        onLayout={[Function]}
                        onMomentumScrollBegin={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        renderScrollComponent={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={[]}
                        testID="transaction-flatlist"
                        viewabilityConfigCallbackPairs={[]}
                        waitFor={
                          [
                            {
                              "current": null,
                            },
                            {
                              "current": null,
                            },
                          ]
                        }
                      >
                        <View />
                      </RCTScrollView>
                    </View>
                  </RCTScrollView>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="ManageTransaction"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
